app/controllers/users_controller.rb

```diff
@@ -4,34 +4,54 @@ class UsersController < ApplicationController
   before_action :require_sign_in
   load_and_authorize_resource
   load_and_authorize_resource :vita_partner, collection: [:edit, :update], parent: false
-
+  before_action :authorize_vita_partner_updates, only: :update
   layout "admin"
 
-  def profile
-  end
+  def profile;end
 
   def index
     @users = @users.includes(:vita_partner)
   end
 
   def edit
+    @user.vita_partner_id = @user.memberships.find_by(role: "member")&.vita_partner_id
+    @user.supported_organization_ids = @user.memberships.where(role: "lead").pluck(:vita_partner_id)
   end
 
   def update
-    authorize!(:manage, @vita_partners.find(user_params[:vita_partner_id]))
-    return render :edit unless @user.update(user_params)
-
+    authorize_vita_partner_updates
+    return render :edit unless update_user
     redirect_to edit_user_path(id: @user), notice: I18n.t("general.changes_saved")
   end
 
   private
+  # To leave the view as-is but allow the ability to save as memberships, we're doing the simplest thing possible:
+  # blowing away and recreating memberships on every update.
+  def update_user
+    ApplicationRecord.transaction do
+      @user.memberships.destroy_all && @user.update!(user_params.merge(memberships_attributes: memberships))
+    end
+
+  rescue ActiveRecord::RecordInvalid
+    logger.info("User update unexpectedly failed. Rolling back transaction.")
+  end
+
+  def memberships
+    supported_org_ids = (params[:user][:supported_organization_ids] || []).reject(&:empty?)
+    memberships = supported_org_ids.map { |so| { user_id: @user.id, vita_partner_id: so, role: "lead"} }
+    memberships.push({user_id: @user.id, vita_partner_id: params[:user][:vita_partner_id], role: "member"}) if params[:user][:vita_partner_id].present?
+    memberships
```

Comment:
I don't think this `supported_organizations` approach to creating many to many relationships to explicitly manage each user's access to each organization is sustainable. We would need a symmetrical process in the controllers for creating child organizations. Furthermore, we do not yet have greeters (which supported organizations was originally designed for).
If we only have clear hierarchical relationships between partners and the orgs that users should be able to edit (I can manage any child of my org) then we should rely on the relationships between orgs to manage access to sub organizations and their clients.

I'd recommend deleting the portion of this form that handles supported organizations and to remove supported organizations as part of this refactor.

Reply:
This is not sustainable, and this is not how we were planning on managing relationships in the long term. It's basically just maintaining the current implementation. The plan is to replace with the ability to add a new organization with new role and remove memberships from this page. Check out the description for why Alex and I decided that the UI refactor was out of scope for this PR! We can remove the supported organizations altogether from the UI if you'd prefer, but Asheesh had implemented something that used supported_organizations to create "leads" of multiple organizations and we wanted to maintain that behavior while adding explicit role.